packages/parameters/tests/unit/AppConfigProvider.test.ts

```diff
@@ -43,6 +49,8 @@ describe('Class: AppConfigProvider', () => {
           serviceId: 'AppConfigData',
         })
       );
+
+      expect(userAgentSpy).toHaveBeenCalled();
```

Comment:
The `client` inside the provider is public and can be accessed via `provider.client`. This means you should be able to test like you did in the `commons` module by doing:

```ts
expect(provider.client.middlewareStack.identify()).toContain(
  'addPowertoolsToUserAgent: POWERTOOLS,USER_AGENT'
);
```

I think that would be preferable to changing the import (`* as ..`) and selecting the module in the spy.

Reply:
> I think that would be preferable to changing the import (* as ..) and selecting the module in the spy. 

Can you elaborate? Curious to hear about the benefits. 

I want to avoid to test the internals of the `commons` module and the dependencies behaviour. Imho, we should only check the call has happened downstream, otherwise we replicate the assertion logic in different places which makes it harder to change in the future. For instance, if we rename the middleware from `addPowertoolsUserAgent` to something else, we'd need to revisit every utility that called this method.

Reply:
Based on the way that the Jest spy is written, if we rename `addPowertoolsUserAgent` to something else the test would also break:

```ts
const userAgentSpy = jest.spyOn(
  UserAgentMiddleware,
  'addUserAgentMiddleware' // if function name changes, this spy also breaks
);
```

However I see your point and agree. I tend to avoid setting spies on imports but I can't think of a better way of doing it without reworking the constructor (which si not worth it), so let's leave it this way!